Remotion/Data/DomainObjects.UnitTests/Persistence/Rdbms/Parameters/SerializedObjectIDDataParameterDefinitionTest.cs

```diff
@@ -0,0 +1,170 @@
+            string.Format("Parameter '{0}' has type '{2}' when type '{1}' was expected.", "storageTypeInformation.DotNetType", typeof(string), typeof(int)),
+            "storageTypeInformation.DotNetType"));
+  }
+
+  [Test]
+  public void GetParameterValue ()
+  {
+    var storageTypeInformationMock = new Mock<IStorageTypeInformation>();
+    var dummyValue = DomainObjectIDs.Order1;
+
+    var dummyStringResult = $"Order|{Guid.NewGuid()}|System.Guid";
+
+    storageTypeInformationMock
+        .Setup(_ => _.ConvertToStorageType(dummyValue.ToString()))
+        .Returns(dummyStringResult);
+
+    storageTypeInformationMock.Setup(_ => _.DotNetType).Returns(typeof(string));
+
+    var serializedObjectIDDataParameterDefinition = new SerializedObjectIDDataParameterDefinition(storageTypeInformationMock.Object);
+
+    var result = serializedObjectIDDataParameterDefinition.GetParameterValue(dummyValue);
+
+    Assert.That(result, Is.EqualTo(dummyStringResult));
+    storageTypeInformationMock.Verify();
+  }
+
+  [Test]
+  public void GetParameterValue_WithNull ()
+  {
+    var storageTypeInformationStub = new Mock<IStorageTypeInformation>();
+    object dummyValue = null;
+
+    storageTypeInformationStub
+        .Setup(_ => _.ConvertToStorageType(dummyValue))
+        .Returns(DBNull.Value);
+    storageTypeInformationStub.Setup(_ => _.DotNetType).Returns(typeof(string));
+
+    var serializedObjectIDDataParameterDefinition = new SerializedObjectIDDataParameterDefinition(storageTypeInformationStub.Object);
+
+    var result = serializedObjectIDDataParameterDefinition.GetParameterValue(dummyValue);
+
+    Assert.That(result, Is.EqualTo(DBNull.Value));
+  }
+
+  [Test]
+  public void GetParameterValue_WithInvalidValue_ThrowsArgumentException ()
+  {
+    var storageTypeInformationStub = new Mock<IStorageTypeInformation>();
+    var dummyValue = "Not an ObjectID";
+
+    storageTypeInformationStub.Setup(_ => _.DotNetType).Returns(typeof(string));
+
+    var serializedObjectIDDataParameterDefinition = new SerializedObjectIDDataParameterDefinition(storageTypeInformationStub.Object);
+
+    Assert.That(
+        () => serializedObjectIDDataParameterDefinition.GetParameterValue(dummyValue),
+        Throws.InstanceOf<ArgumentException>().With.ArgumentExceptionMessageEqualTo(
+            string.Format("Parameter '{0}' has type '{2}' when type '{1}' was expected.", "parameterObject", typeof(ObjectID), typeof(string)),
+            "parameterObject"));
+  }
+
+  [Test]
+  public void CreateDataParameter ()
+  {
+    var commandMock = new Mock<IDbCommand>(MockBehavior.Strict);
+    var dataParameterMock = new Mock<IDbDataParameter>(MockBehavior.Strict);
+
+    var storageTypeInformation = new StorageTypeInformation(
+        typeof(string),
+        "test",
+        DbType.AnsiString,
+        false,
+        -1,
+        typeof(string),
+        new DefaultConverter(typeof(string)));
+
+    var testValue = $"Order|{Guid.NewGuid()}|System.Guid";
```

Comment:
should we use `ObjectIDStringSerializer.Instance.Serialize` instead?
(or `ObectID.ToString()`)
Having the serialization format hardcoded here feels like duplication